inc/class-storefront.php

```diff
@@ -135,23 +135,27 @@ public function widgets_init() {
 				'id'            => 'sidebar-1',
 				'description'   => ''
 			);
-
+			
 			$sidebar_args['header'] = array(
 				'name'        => __( 'Below Header', 'storefront' ),
 				'id'          => 'header-1',
 				'description' => __( 'Widgets added to this region will appear beneath the header and above the main content.', 'storefront' ),
 			);
 
-			$footer_widget_regions = apply_filters( 'storefront_footer_widget_regions', 4 );
 
-			for ( $i = 1; $i <= intval( $footer_widget_regions ); $i++ ) {
-				$footer = sprintf( 'footer_%d', $i );
+			$rows    = intval( apply_filters( 'storefront_footer_widget_rows', 2 ) );
+			$regions = intval( apply_filters( 'storefront_footer_widget_regions', 4 ) );
 
-				$sidebar_args[ $footer ] = array(
-					'name'        => sprintf( __( 'Footer %d', 'storefront' ), $i ),
-					'id'          => sprintf( 'footer-%d', $i ),
-					'description' => sprintf( __( 'Widgetized Footer Region %d.', 'storefront' ), $i )
-				);
+			for ( $row = 1; $row <= $rows; $row++ ) {
+				for ( $region = 1; $region <= $regions; $region++ ) {
+					$footer = sprintf( 'footer_%1$d_%2$d', $row, $region );
+
+					$sidebar_args[ $footer ] = array(
+						'name'        => sprintf( __( 'Footer Row-%1$d Region-%2$d', 'storefront' ), $row, $region ),
+						'id'          => sprintf( 'footer-%1$d-%2$d', $row, $region ),
```

Comment:
We may have a problem with widget ID's here - if it's a problem in first place.
After update, users will loose their footer settings though their widgets will be put on side in "Inactive Sidebars" sections.

If you consider it a problem, there is a simple workaround that is ready for commit.
Keep the old naming convention, eg:
- `'footer-1'`
- `'footer-2'`
- `'footer-3'`
- `'footer-4'`
- `'footer-5'`
- and so on...

... This way:

```php
'id' => sprintf( 'footer-%d', $region + $regions * ( $row - 1 ) );
```
With the same approach in the template function